features/Dockerfile

```diff
@@ -0,0 +1,74 @@
+# syntax = docker/dockerfile:1.5
+# Used only for running tests using tox, see ../tox.ini
+ARG PG_MAJOR
+ARG PGHOME=/home/postgres
+ARG PGDATA=$PGHOME/data
+ARG LC_ALL=C.UTF-8
+ARG LANG=C.UTF-8
+
+FROM postgres:${PG_MAJOR}
+
+ARG PGHOME
+ARG PGDATA
+ARG LC_ALL
+ARG LANG
+
+ENV ETCDVERSION=3.3.13
+
+USER root
+RUN set -ex \
+    && apt-get update \
+    && apt-get reinstall init-system-helpers \
+    && apt-get install -y \
+      python3-pip \
+      python3-dev \
+      rsync \
+      curl \
+      gcc \
+      golang \
+      jq  \
+      locales  \
+      sudo \
+      busybox \
+      net-tools \
+      iputils-ping \
+    && rm -rf /var/cache/apt \
+    && python3 -m pip install --no-cache-dir tox \
+    \
+    && mkdir -p "$PGHOME" \
+    && sed -i "s|/var/lib/postgresql.*|$PGHOME:/bin/bash|" /etc/passwd \
+    && chown -R postgres:postgres /var/log /home/postgres \
+    \
+    # Download etcd \
+    && ETCDURL="https://github.com/coreos/etcd/releases/download/v$ETCDVERSION" \
+    && curl -sL "$ETCDURL/etcd-v$ETCDVERSION-linux-$(dpkg --print-architecture).tar.gz" \
+       | tar xz -C /usr/local/bin --strip=1 --wildcards --no-anchored etcd etcdctl
+
+
+# This Dockerfile syntax only works with docker buildx and the syntax
+# line at the top of this file.
+COPY <<EOF /tox-wrapper.sh
+#!/usr/bin/env bash
+set -ex
+copy_output() {
+    if [[ -d "$PGHOME/src/features/output" && /src/features ]] ;then
+        cp -a "$PGHOME/src/features/output" "/src/features/output-\$HOSTNAME"
+        find "/src/features/output-\$HOSTNAME" -type f -exec chmod 666 {} \\;
+        find "/src/features/output-\$HOSTNAME" -type d -exec chmod 777 {} \\;
+    fi
+}
+trap 'copy_output' SIGTERM
+# For architectures such as aarch we need to get the respective GOARCH
+# so we can tell etcd we're ok with running an unsupported architecture.
+export ETCD_UNSUPPORTED_ARCH=$(go env GOARCH)
+runuser -u postgres -- rsync -a --exclude=.tox /src/ "$PGHOME/src/"
+cd "$PGHOME/src"
+runuser -u postgres -w ETCD_UNSUPPORTED_ARCH -- "\$@" &
+wait $?
```

Comment:
The var there should have been `$!` not `$?` (which is the exit status of the prev cmd!)
I'm not sure why that didn't cause me a problem before.

However, with your previous suggestion of `trap EXIT` we could make it simpler and not background the command or make it `wait`. It was mainly there in case of doing something else after tox, which I did before but that's gone now. This make it cleaner I think.

I'm just testing it still works as expected.

Reply:
It explains a lot:) sorry, wasn't careful enough

Reply:
This should be fixed now

Reply:
Please see my other comment on refactoring this.